Menus/SettingsInputMappingWizardGUI.cpp

```diff
@@ -0,0 +1,1065 @@
+				} else {
+					m_ConfigStep--;
+				}
+				m_ConfigStepChange = true;
+			} else if (guiEvent.GetControl() == m_WizardManualConfigScreen.NextConfigStepButton) {
+				g_GUISound.ButtonPressSound()->Play();
+				m_ConfigStep++;
+				m_ConfigStepChange = true;
+			} else if (guiEvent.GetControl() == m_WizardManualConfigScreen.ResetConfigButton) {
+				g_GUISound.ButtonPressSound()->Play();
+				ResetManualConfigSequence();
+			} else if (guiEvent.GetControl() == m_WizardManualConfigScreen.DiscardOrApplyConfigButton) {
+				g_GUISound.ButtonPressSound()->Play();
+				if (m_ConfigFinished) {
+					ApplyManuallyConfiguredScheme();
+				} else {
+					if (m_ConfiguringDeviceIsGamepad) {
+						ShowPresetSelectionScreen();
+					} else {
+						SetEnabled(false);
+					}
+				}
+			}
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void SettingsInputMappingWizardGUI::HandlePresetSelectScreenInputEvents(GUIEvent &guiEvent) {
+		if (guiEvent.GetType() == GUIEvent::Command) {
+			if (guiEvent.GetControl() == m_WizardPresetSelectScreen.CloseWizardButton) {
+				g_GUISound.ButtonPressSound()->Play();
+				SetEnabled(false);
+			} else if (guiEvent.GetControl() == m_WizardPresetSelectScreen.StartConfigDPadTypeButton) {
+				g_GUISound.ButtonPressSound()->Play();
+				m_ConfiguringGamepadType = GamepadType::DPad;
+				ShowManualConfigScreen();
+			} else if (guiEvent.GetControl() == m_WizardPresetSelectScreen.StartConfigAnalogDSTypeButton) {
+				g_GUISound.ButtonPressSound()->Play();
+				m_ConfiguringGamepadType = GamepadType::AnalogDualShock;
+				ShowManualConfigScreen();
+			} else if (guiEvent.GetControl() == m_WizardPresetSelectScreen.StartConfigAnalogXBTypeButton) {
+				g_GUISound.ButtonPressSound()->Play();
+				m_ConfiguringGamepadType = GamepadType::AnalogXbox;
+				ShowManualConfigScreen();
+			} else if (guiEvent.GetControl() == m_WizardPresetSelectScreen.PresetSelectSNESButton) {
+				ApplyGamepadInputPreset(GamepadType::DPad);
+			} else if (guiEvent.GetControl() == m_WizardPresetSelectScreen.PresetSelectDS4Button) {
+				ApplyGamepadInputPreset(GamepadType::AnalogDualShock);
+			} else if (guiEvent.GetControl() == m_WizardPresetSelectScreen.PresetSelectXB360Button) {
+				ApplyGamepadInputPreset(GamepadType::AnalogXbox);
+			}
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void SettingsInputMappingWizardGUI::HandleManualConfigSequence() {
+		if (m_ConfigStepChange) {
+			ConfigWizardSteps configuringDeviceSteps = ConfigWizardSteps::NoConfigSteps;
+
+			switch (m_ConfiguringDevice) {
+				case InputDevice::DEVICE_KEYB_ONLY:
+					configuringDeviceSteps = ConfigWizardSteps::KeyboardConfigSteps;
+					break;
+				case InputDevice::DEVICE_MOUSE_KEYB:
+					configuringDeviceSteps = ConfigWizardSteps::MouseAndKeyboardConfigSteps;
+					break;
+				default:
```

Comment:
Do you honestly prefer this for something that will never happen?
```c++
switch (m_ConfiguringDevice) {
	case InputDevice::DEVICE_KEYB_ONLY:
		configuringDeviceSteps = ConfigWizardSteps::KeyboardConfigSteps;
		break;
	case InputDevice::DEVICE_MOUSE_KEYB:
		configuringDeviceSteps = ConfigWizardSteps::MouseAndKeyboardConfigSteps;
		break;
	case InputDevice::DEVICE_GAMEPAD_1:
	case InputDevice::DEVICE_GAMEPAD_2:
	case InputDevice::DEVICE_GAMEPAD_3:
	case InputDevice::DEVICE_GAMEPAD_4:
		configuringDeviceSteps = (m_ConfiguringGamepadType == GamepadType::DPad) ? ConfigWizardSteps::DPadConfigSteps : ConfigWizardSteps::DualAnalogConfigSteps;
		break;
	default:
		RTEAbort("Invalid InputDevice passed to SettingsInputMappingWizardGUI::HandleManualConfigSequence!");
		break;
}
```

Reply:
Ohhhh, right, each one is another input device.
Hmmm, compare to
```cpp
switch (m_ConfiguringDevice) {
	case InputDevice::DEVICE_KEYB_ONLY:
		configuringDeviceSteps = ConfigWizardSteps::KeyboardConfigSteps;
		break;
	case InputDevice::DEVICE_MOUSE_KEYB:
		configuringDeviceSteps = ConfigWizardSteps::MouseAndKeyboardConfigSteps;
		break;
	default:
		configuringDeviceSteps = (m_ConfiguringGamepadType == GamepadType::DPad) ? ConfigWizardSteps::DPadConfigSteps : ConfigWizardSteps::DualAnalogConfigSteps;
		break;
}
```

I dunno, your original one is obviously more compact but the longer one is a little more obvious with what's going on, and follows the general safety rule of thumb for switches. Without it, if someone comes in here and does something dumb, it could screw up quietly instead of making it very clear what went wrong. But the odds of that happening and not getting caught in review are minuscule, so I don't think it's really worth worrying about.

I guess I'll leave it to you, change it if you want to or don't if you don't want to.